app/main.py

```diff
@@ -1,2 +1,144 @@
+            elif target[0] != key:
+                self._resize()
+                index = key_hash % self.capacity
+
+        while True:
+            target = self.hash_table[index]
+            if target is None or target == "del":
+                self.hash_table[index] = [key, key_hash, value]
+                self.length += 1
+                break
+            elif target[1] == key_hash and target[0] == key:
+                self.hash_table[index][2] = value
+                break
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key: Any) -> Any:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        start_index = index
+
+        while True:
+            target = self.hash_table[index]
+            if target is None:
+                raise KeyError(f"Key {key} does not exist")
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                return target[2]
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                raise KeyError(f"Key {key} does not exist")
+
+    def clear(self) -> None:
+        self.hash_table = [None] * self.capacity
+        self.length = 0
+
+    def __delitem__(self, key: Any) -> None:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        start_index = index
+
+        while True:
+            target = self.hash_table[index]
+            if target is None:
+                raise KeyError(f"Key {key} does not exist")
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                self.hash_table[index] = "del"
+                self.length -= 1
+                break
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                raise KeyError(f"Key {key} does not exist")
+
+    def get(self, key: Any, default: Any = None) -> Any:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        start_index = index
+
+        while True:
+            target = self.hash_table[index]
+            if target is None:
+                return default
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                return target[2]
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                return default
+
+    def pop(self, key: Any, default: Any = "raise") -> None:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        start_index = index
+
+        while True:
+            target = self.hash_table[index]
+            if target is None:
+                if default == "raise":
+                    raise KeyError(f"Key {key} does not exist")
+                return default
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                value = target[2]
+                self.hash_table[index] = "del"
```

Comment:
The `pop` method should immediately remove the element. Your current solution is incorrect.

Reply:
Used` __delitem__` method